### PR TITLE
Add support for arbitrary extensions to be bundled in XI/XM apps (#9048)

### PR DESCRIPTION
**Pull request by chamons**
- This commit adds a hook, "AdditionalAppExtensions", to the msbuild to allow
extensions written in other languages, such as Swift, to be embedded and signed in an
Xamarin App bundle easily.
- Example:
<AdditionalAppExtensions Include="$(MSBuildProjectDirectory)/../../native">
	<Name>NativeTodayExtension</Name>
	<BuildOutput Condition="'$(Platform)' == 'iPhone'">build/Debug-iphoneos</BuildOutput>
	<BuildOutput Condition="'$(Platform)' == 'iPhoneSimulator'">build/Debug-iphonesimulator</BuildOutput>
</AdditionalAppExtensions>

Cherry pick of https://github.com/xamarin/xamarin-macios/pull/9048 to xcode12.

### CODE REVIEW

File: msbuild/Xamarin.Shared/Xamarin.Shared.targets [hunk 1]
```diff
@@ -60,7 +60,32 @@ Copyright (C) 2018 Microsoft. All rights reserved.
...
+			<_ResolvedAppExtensionReferences Include="%(AdditionalAppExtensions.Identity)/%(AdditionalAppExtensions.BuildOutput)/%(AdditionalAppExtensions.Name).appex" />
+
+			<_AppExtensionCodesignProperties Include="%(AdditionalAppExtensions.Name).appex">
+				<Entitlements Condition="Exists('$(_ExtensionEntitlementPath)')">
+					$(_ExtensionEntitlementPath)
+				</Entitlements>
+				<SigningKey>$(CodesignKey)</SigningKey>
+				<Keychain>$(CodesignKeychain)</Keychain>
+				<DisableTimestamp>False</DisableTimestamp>
+				<DisableTimestamp Condition="'$(_SdkIsSimulator)' == 'true' Or '$(_BundlerDebug)' == 'true'">True</DisableTimestamp>
```
Comment (rolfbjarne) [inline msbuild/Xamarin.Shared/Xamarin.Shared.targets:81]: `_BundlerDebug` isn't in the xcode12 branch, not sure what's the best thing to do :/

Comment (chamons) [reply msbuild/Xamarin.Shared/Xamarin.Shared.targets:81]: Could I use both the XM and XI specific variables A OR B OR C?

Comment (rolfbjarne) [reply msbuild/Xamarin.Shared/Xamarin.Shared.targets:81]: Yes, I think that would work. Ugly, but it'll be temporary.

**Comment by chamons** [comment]
Great, the test API also changed:

```
11:07:44          ProjectsTests/EmbeddedExtension.cs(26,176): error CS1739: The best overload for 'Execute' does not have a parameter named 'environment_variables' [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/msbuild/tests/Xamarin.iOS.Tasks.Tests/Xamarin.iOS.Tasks.Tests.csproj]
11:07:44          ProjectsTests/EmbeddedExtension.cs(29,172): error CS1739: The best overload for 'Execute' does not have a parameter named 'environment_variables' [/Users/builder/jenkins/workspace/xamarin-macios-pr-builder/msbuild/tests/Xamarin.iOS.Tasks.Tests/Xamarin.iOS.Tasks.Tests.csproj]
11:07:44 
```

**Comment by chamons** [comment]
I'm marking this cherry-pick as ready as I think I got the merge issues from main.

If it misbehaves just ignore it and I'll fix when I get back.

**Comment by chamons** [comment]
Blocked on https://github.com/xamarin/xamarin-macios/pull/9162 landing and re-merge

**Pull request merged**